crypto-app/src/App.tsx

```diff
@@ -1,7 +1,32 @@
 import Navbar from "@/scenes/navbar"
+import { useEffect, useState } from "react"
+import { SelectedPage } from "./shared/types";
+
+
 function App() {
+  const [selectedPage, setSelectedPage] = useState<SelectedPage>(
+    SelectedPage.Home
+    );
+  const [isTopOfPage, setIsTopOfPage] = useState<boolean>(true);
+
+  useEffect(() => {
+    const handleScroll = () => {
+      if(window.scrollY === 0) {
+        setIsTopOfPage(true)
+        setSelectedPage(SelectedPage.Home);
+      }
+      if(window.scrollY !== 0) setIsTopOfPage(false);
```

Comment:
You could simplify this:
```suggestion
      if(window.scrollY !== 0) {
        return setIsTopOfPage(false)
      }
        setIsTopOfPage(true)
        setSelectedPage(SelectedPage.Home);;
```

early guard-clause, logic is simpler